### PR TITLE
Shrink size of Klaro & ban wildcard imports.

### PR DESCRIPTION
**Pull request by tdonohue**
Related to #1921 

## References
This PR continues to try to decrease the size of `main.js`.
* This PR includes the changes in #1903 (so it is smaller than it appears)

**This PR decreases the GZipped size of `main.js` by an additional ~15KB.** (on top of the ~66KB decrease of #1903, totaling a ~81KB decrease).  This measurement was determined by running the following on `main` vs this PR:
  * `yarn build:stats`
  * `webpack-bundle-analyzer .\dist\browser\stats.json`

## Description
This PR updates the following:
* Bans usage of wildcard imports (except where required).  This ensures these wildcard imports don't accidentally import an entire library when all we need is a few features of the library.  (This change doesn't appear to have shrunk `main.js` much, but it's a safety measure to avoid future growth of `main.js`)
    * For example, this is **not** allowed: `import * as fs from 'fs';`
    * Instead, you should do something like: `import { existsSync, writeFileSync } from 'fs';`
* Replaced wildcard imports where possible to do so.  (For some mini-libraries this is not possible. But they are exceptions to the new rule.)
* Upgrade to latest Klaro
* Switch to importing `klaro-no-translations`, which excludes the built in translations in Klaro & decreases the size of Klaro.  This is because we are already translating much of Klaro in our existing `src/assets/i18n/*.json5` files.  Added a few more keys to ensure all of Klaro is translated in DSpace.

## Instructions for Reviewers
* Test Klaro cookie popup.  Use Incognito mode to verify the Klaro popup is fully translated.  Switch to a different language and verify it is still _mostly_ translated (there may be a few English words still if the `*.json5` translation pack you select is missing a few keys).
* Review code changes.

### CODE REVIEW

**Comment by tdonohue** [comment]
@artlowel : Assigning to you for review as you volunteered to look at shrinking main.js.  Please feel free to delegate if you wish.

File: src/app/shared/mocks/dspace-rest/endpoint-mocking-rest.service.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { RestRequestMethod } from '../../../core/data/rest-request-method';
 import { RawRestResponse } from '../../../core/dspace-rest/raw-rest-response.model';
 import { DspaceRestService, HttpOptions } from '../../../core/dspace-rest/dspace-rest.service';
 import { MOCK_RESPONSE_MAP, ResponseMapMock } from './mocks/response-map.mock';
+// eslint-disable-next-line import/no-namespace
 import * as URL from 'url-parse';
```
Comment (ybnd) [inline src/app/shared/mocks/dspace-rest/endpoint-mocking-rest.service.ts:11]: This import can be left out
The usage below will just work with native `new URL(...)`

File: src/app/core/index/index.selectors.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { hasValue, isNotEmpty } from '../../shared/empty.util';
 import { coreSelector } from '../core.selectors';
 import { URLCombiner } from '../url-combiner/url-combiner';
 import { IndexState, MetaIndexState } from './index.reducer';
+// eslint-disable-next-line import/no-namespace
 import * as parse from 'url-parse';
```
Comment (ybnd) [inline src/app/core/index/index.selectors.ts:7]: This import can be left out, we can replace
* `parse(...)` → `new URL(...)`
* `parsed.query` → `parsed.search`

**Review by ybnd — CHANGES_REQUESTED** [review]
Confirmed that
* Klaro still works as before, including i18n
* the new ESLint rule works

Looks like [url-parse](https://www.npmjs.com/package/url-parse) is not necessary anymore; we should be able to replace it with [the native URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL).

Checked this out real quick & it seems to work fine, server- and client-side.

File: src/app/shared/mocks/dspace-rest/endpoint-mocking-rest.service.ts (hunk 1 above)
Comment (tdonohue) [reply src/app/shared/mocks/dspace-rest/endpoint-mocking-rest.service.ts:11]: Good find, @ybnd .  I'll get these usages of url-parse cleaned up and verify they work for me.

**Comment by tdonohue** [comment]
@ybnd : I've addressed your prior feedback.  If you could give this a quick rereview, I'd appreciate it!

**Review by ybnd — APPROVED** [review]
Changes look good & everything works as it should!

**Pull request merged**